x-pack/plugins/kubernetes_security/public/components/container_name_widget/container_name_row.tsx

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { ReactNode, useState } from 'react';
+import { EuiFlexItem, EuiText } from '@elastic/eui';
+import { useStyles } from './styles';
+
+export interface ContainerNameRowDeps {
+  name: string;
+  index?: number;
+  filterButtonIn?: ReactNode;
+  filterButtonOut?: ReactNode;
+}
+
+export const ROW_TEST_ID = 'kubernetesSecurity:containerNameSessionRow';
+
+export const ContainerNameRow = ({
+  name,
+  index,
+  filterButtonIn,
+  filterButtonOut,
+}: ContainerNameRowDeps) => {
+  const [hoveredFilter, setHoveredFilter] = useState<number | null>(null);
```

Comment:
this can just be a boolean `isHover`. No need to keep track of index in this component since each component row handles hover actions independently